transport/http/dial.go

```diff
@@ -286,10 +301,9 @@ func getEndpoint(settings *internal.DialSettings, clientCertSource cert.Source)
 }
 
 func getMTLSMode() string {
-	mode := os.Getenv("GOOGLE_API_USE_MTLS")
+	mode := os.Getenv("GOOGLE_API_USE_MTLS_ENDPOINT")
```

Comment:
I think we should still support the old variable here. This feature has been released and could be in use, right? I recommend honoring the new one first, and fallback to the old if not set. Then mention in the commit that we are officially deprecating the use of the old environment variable.

Reply:
Good point. Done.